context.go

```diff
@@ -169,6 +169,7 @@ func (c *Context) LookupTypeArray(inner Type) *TypeArray {
 }
 
 func (c *Context) LookupTypeUnion(types []Type) *TypeUnion {
+	SortTypes(types)
```

Comment:
Nit: I don't think SortTypes is worth adding.
```suggestion
	sort.SliceStable(types, func(i, j int) bool {
		return CompareTypes(types[i], types[j]) < 0
	})
```